lib/State.re

```diff
@@ -51,20 +51,22 @@ module Make = (StateContextImpl: StateContext) => {
       curr;
     };
 
-  let pushNewState: (t, 'a) => updateFunction('a) =
+  let pushNewState: (t, 'a) => getterAndUpdater('a) =
     (state: t, currentVal: 'a) => {
       let updatedVal: ref(Object.t) = ref(Object.to_object(currentVal));
       state.newState = List.append(state.newState, [updatedVal]);
-      let ret: updateFunction('a) =
+      let ret: getterAndUpdater('a) = (
+        () => Object.of_object(updatedVal^),
```

Comment:
I am not 100% sure this is necessary... but I added it just to be sure, for cases when there are two actions being dispatched in the same render.